source/docs/example-projects/halloween-treasure-hunt.rst

```diff
@@ -0,0 +1,239 @@
+==========================
+1. Halloween Treasure Hunt
+==========================
+
+.. contents:: Table of Contents
+    :local:
+
+Introduction
+============
+
+In this first example project, we'll start with the basics. The following topics will be covered:
+
+* Denizen objects and general syntax.
+* Browsing through the documentation.
+* Reloading scripts and debug modes.
+* World script containers, events and context tags.
+* Common script commands and arguments.
+* Constructing tags with simple bases and modifiers.
+* Basic usage of foreach loops.
+
+If you are not familiar with any of the concepts listed, we suggest revisiting the following tutorial documents:
+
+* :doc:`Section 1.3 - dScript Format </docs/getting-started/dscript-format>`
+* :doc:`Section 2.1 - An Introduction to Tags </docs/basics-of-scripting/an-introduction-to-tags>`
+* :doc:`Section 2.2 - The Basics of Denizen Commands </docs/basics-of-scripting/basics-of-denizen-commands>`
+* :doc:`Section 2.7 - Your First World Script </docs/basics-of-scripting/your-first-world-script>`
+* **[TODO: Add FOREACH section reference]**
+
+This will help you make a better use of this example project, and follow its contents successfully.
+
+.. note::
+    It is also recommended to have the `meta documentation`__ open in another tab while scripting this project, as 
+    we'll need to check the syntax of events, commands and tags very often.
+
+.. __: https://one.denizenscript.com/denizen/logs
+    
+1. First Steps
+==============
+
+Now let's jump (or slowly crawl) into the scripting process. In our case, we want to make a **Halloween Treasure 
+Hunt** event in our *Hub world*. We plan on manually placing some hidden pumpkins ourselves, and reward players for 
+finding (and left clicking) them. First of all, we'll need to *create a new script file* (for example 
+``halloween-treasure-hunt.yml``) and build a world  script container which, as we know, will listen to events 
+happening within our server.
+
+Our file with the script container looks like this:
+
+.. code-block:: dscript
+    :name: figureDIY_1_1
+    :linenos:
+
+    halloween_treasure_hunt:
+    type: world
+    events:
+
+.. rst-class:: figurecaption
+
+**Figure 1.1** Our starting world script container
+
+Now we need to find an event that fits our case. Looking through the event list documentation, `on player clicks 
+block`__ looks like the best match. We just have to adjust it for our specific use case by adding optional arguments. 
+According to the event syntax, it accepts a click type, a material and an area switch, which will just be ``left``, 
+``carved_pumpkin`` and ``in:Hub`` respectively.
+
+.. __: https://one.denizenscript.com/denizen/evts/clicks%20block
+
+.. note::
+    This tutorial assumes our *Hub world* is in fact called ``Hub``.
```

Comment:
This note might be a little out of place. It may be better to just mention in the paragraph at line 40 that this can be done in any world, but that you're using a world named "Hub" as an example world that the user may use.